galsim/phase_psf.py

```diff
@@ -1085,6 +1094,14 @@ def _finalize(self, flux, suppress_warning):
         self.img *= (flux / (self.img.sum() * self.scale**2))
         self.img = galsim.ImageD(self.img.astype(np.float64), scale=self.scale)
 
+        if self.max_size is not None and self.max_size < self.img.scale*self.img.array.shape[0]:
+            center = self.img.center()
+            bds = galsim.BoundsI(center)
+            half_max_size = int(np.ceil(0.5*self.max_size/self.img.scale))
+            bds += center + galsim.PositionI(-half_max_size, -half_max_size)
+            bds += center + galsim.PositionI(half_max_size, half_max_size)
+            self.img = self.img[bds]
```

Comment:
Doing this after the calculation of the full size image removes a big part of the benefit of this parameter.  The point was mostly to save time during the calculation of the image -- to not make so large an fft when building the optical PSF if we don't actually need the full image.  I don't think the running time is much different if you shrink the image at this point for the input to the InterpolatedImage.

Is there some reason that you think that was invalid and switched to doing it this way?  If so, maybe we just ought to remove this parameter as an option at all, since I don't think it's very helpful this way.